src/odemis/gui/cont/project.py

```diff
@@ -283,22 +279,22 @@ def __init__(self, name, roc_2, roc_3, colour, tab_data, project_panel, viewport
         # Read the overlap from the acquisition configuration
         acqui_conf = conf.get_acqui_conf()
 
-        self.model = FastEMROA(name, acqstream.UNDEFINED_ROI, roc_2, roc_3,
+        self.model = FastEMROA(name, roc_2, roc_3,
                                self._tab_data.main.asm, self._tab_data.main.multibeam,
                                self._tab_data.main.descanner, self._tab_data.main.mppc,
                                acqui_conf.overlap)
-        self.model.coordinates.subscribe(self._on_coordinates)
         self.model.roc_2.subscribe(self._on_roc)
         self.model.roc_3.subscribe(self._on_roc)
 
         # The panel is not created on initialization to allow for cancellation of the ROA creation
         # (cf discussion in FastEMProjectController), create panel with .create_panel().
         self.panel = None
 
-        logging.debug("Creating overlay for ROA '%s'.", name)
-        canvas = self._viewport.canvas
-        self.overlay = canvas.add_roa_overlay(self.model.coordinates, colour)
-        self.overlay.active.subscribe(self._on_overlay_active)
+        self.shape = shape
+        self.shape.colour = conversion.hex_to_frgba(colour)
+        self.shape.points.subscribe(self._on_shape_points)
+        self.shape.selected.subscribe(self._on_shape_selected)
+        self.shape.position.subscribe(self._find_closest_scintillator)
```

Comment:
`position` (often) changes when `points` changes. I'd suggest to just listen to .points, and call `_find_closest_scintillator` from there.